### ISSUE TITLE
Cannot update and save question - you don’t have permission to see that

### ISSUE THREAD

**Issue by etoulas** [metabase/metabase#43807]
### Describe the bug

When I try to update and save an innocent SQL question, I get the error:

> Sorry, you don’t have permission to see that.

![image](https://github.com/metabase/metabase/assets/4390468/1c1ae527-63ed-4693-8035-11a278b444a2)

### To Reproduce

I have admin permissions and one question saved in Our Analytics.

When I modify it and run the query it works as expected.
When saving the question, overwriting the existing one, it will immediately fail.

The SQL is pretty innocent and runs against Athena. (shortened)

I simply add a few more rows to have up-to-date fx rates.

```sql
WITH CurrencyConversion (month, currency, rate) AS (
    SELECT month, currency, rate
    FROM (VALUES

(timestamp '2024-01-01', 'CHF', 0.9368227273),
(timestamp '2024-02-01', 'CHF', 0.9462190476),

(timestamp '2024-01-01', 'EUR', 1.0),
(timestamp '2024-02-01', 'EUR', 1.0)

    ) AS CurrencyConversion (month, currency, rate)
)
SELECT *
FROM CurrencyConversion
```

Strangely enough, I have a similar questions right next to it and no problems updating it.

### Expected behavior

Saving a change to a question should work especially considering I'm an admin.

### Logs

Metabase logs:

```
[6114bda0-916e-410b-9d52-bb99214b9d60] 2024-06-07T11:41:29+02:00 DEBUG metabase.server.middleware.log GET /api/alert/question/53 200 1.8 ms (1 DB calls) App DB connections: 2/15 Jetty threads: 4/50 (3 idle, 0 queued) (131 total active threads) Queries in flight: 0 (0 queued)
[6114bda0-916e-410b-9d52-bb99214b9d60] 2024-06-07T11:41:31+02:00 DEBUG metabase.server.middleware.log POST /api/card/53/query 202 [ASYNC: completed] 710.0 ms (15 DB calls) App DB connections: 3/15 Jetty threads: 2/50 (4 idle, 0 queued) (131 total active threads) Queries in flight: 0 (0 queued); athena DB 4 connections: 0/1 (0 threads blocked)
```

Chrome console:

![image](https://github.com/metabase/metabase/assets/4390468/fbc8feb8-5962-4837-a0f0-67fb21054693)

### Information about your Metabase installation

```JSON
Hosted on AWS.

Metabase -> Athena -> S3 -> ORC files


{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.23+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.23",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.23+9",
    "os.name": "Linux",
    "os.version": "5.10.215-203.850.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "Europe/Zurich"
  },
  "metabase-info": {
    "databases": [
      "athena",
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.13"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-05-09",
      "tag": "v0.49.9",
      "hash": "c0913c7"
    },
    "settings": {
      "report-timezone": "Europe/Zurich"
    }
  }
}
```

### Severity

Blocks updating questions that are used in different places

**Comment by etoulas** [comment]
Another input.

I managed to workaround the problem for that particular question with fx rates:

I used "Format query" in the SQL editor. After that I could save without problem.

It looks like the SQL format has to match some expectation somewhere.

(Side note: in a previous setup that connected to a Postgres DB I sometimes ran into a similar issue if the last line of the query contained a `-- comment`. I thought I mention it, even though it is not exactly the same issue.)
